### DISCUSSION TITLE
tdiary でのウェブアプリケーションフレームワークについて

### DISCUSSION THREAD

**Discussion by Sa2shi** [tdiary/tdiary-core#1045] [Q&A] [answered]
お世話になっております。Ruby初学者です。
表題の件で問合せさせていただきます。

◎質問

tdiary は、「Ruby on Rails」や「Sinatra」といったWebアプリケーションフレームワークは使われていないでしょうか。また、もし使われていない場合、独自に先述のフレームワークを用いて、tdiaryのサーバ構築は可能でしょうか。

○背景
自身のラズパイ環境のtdiaryでのOSS管理の上でSCAツール活用を検討しており、当該ツールの動作設定の上で参考にさせていただきたいため。

**Comment by hsbt** [comment, marked as answer]
tDiary は Rails や Sinatra などのフレームワークは用いておらず、Rack と呼ばれる共通APIを使っています。また、Rails や Sinatra も Rack を用いています。

という前置きをした上で、ご質問への回答ですが、Rails や Sinatra はフレームワークであってサーバーを構築するためのものではないので、tDiary を動かすことはできません。一方で Rack を使ってサーバーを作ることはできるか、という質問であれば可能です。

"Rack サーバー" とか、"Rack puma" などの単語で検索すると情報が得られると思います。

**Reply by Sa2shi** [reply]
ご回答ありがとうございます。イメージ掴めました。大変助かりました。
